runner/src/runner.rs

```diff
@@ -0,0 +1,1127 @@
+			runner.expect_download_path().return_const(mock_path.clone());
+			runner.expect_client_type().return_const(client.clone());
+			let (bin_name, bin_path) = Runner::get_bin_path(
+                &runner,
+                "https://github.com/interlay/interbtc-clients/releases/download/1.17.2/vault-parachain-metadata-kintsugi",
+            )
+                .unwrap();
+			assert_eq!(bin_name, "vault-parachain-metadata-kintsugi");
+			assert_eq!(bin_path, mock_path.join(bin_name));
+		}
+	}
+
+	#[tokio::test]
+	async fn test_runner_delete_downloaded_release() {
+		let tmp = TempDir::new("runner-tests").expect("failed to create tempdir");
+		// Create dummy file
+		let mock_path = tmp.path().join("mock_file");
+		File::create(mock_path.clone()).unwrap();
+
+		let mut runner = MockRunner::default();
+		let downloaded_release = DownloadedRelease {
+			checksum: H256::default(),
+			path: mock_path.clone(),
+			bin_name: String::default(),
+		};
+		runner.expect_downloaded_release().return_const(Some(downloaded_release));
+		runner.expect_set_downloaded_release().return_const(());
+
+		Runner::delete_downloaded_release(&mut runner).unwrap();
+		assert_eq!(mock_path.exists(), false);
+	}
+
+	#[tokio::test]
+	async fn test_runner_terminate_proc_and_wait() {
+		// spawn long-running child process
+		let mut runner = MockRunner::default();
+		runner
+			.expect_child_proc()
+			.returning(|| Some(Command::new("sleep").arg("100").spawn().unwrap()));
+		runner.expect_set_child_proc().return_const(());
+		let pid = Runner::terminate_proc_and_wait(&mut runner).unwrap();
+		let pid_i32: i32 = pid.try_into().unwrap();
+		let s = System::new_all();
+		// Get all running processes
+		let processes = s.processes();
+		// Get the child process based on its pid
+		let child_process = processes.get(&Pid::from(pid_i32));
+
+		assert_eq!(child_process.is_none(), true);
+	}
+
+	#[tokio::test]
+	async fn test_runner_run_binary_with_retry() {
+		let tmp = TempDir::new("runner-tests").expect("failed to create tempdir");
+
+		let mock_executable_path = tmp.path().join("print_cli_input");
+		{
+			let mut file = OpenOptions::new()
+				.read(true)
+				.write(true)
+				.mode(0o700)
+				.create(true)
+				.open(mock_executable_path.clone())
+				.unwrap();
+
+			// Script that prints CLI input to stdout
+			file.write_all(b"#!/bin/bash\necho $@").unwrap();
+
+			file.sync_all().unwrap();
+			// drop `file` here to close it and avoid `ExecutableFileBusy` errors
+		}
+
+		let mut runner = MockRunner::default();
+		let mock_vault_args: Vec<String> = vec![
+			"--bitcoin-rpc-url",
+			"http://localhost:18443",
+			"--bitcoin-rpc-user",
+			"rpcuser",
+			"--bitcoin-rpc-pass",
+			"rpcpassword",
+			"--keyfile",
+			"keyfile.json",
+			"--keyname",
+			"0xa81f76187f1e5d2059f67439c4242a92a5cd66a409579db73f156c6e2aae5102",
+			"--faucet-url",
+			"http://localhost:3033",
+			"--auto-register=KSM=faucet",
+			"--btc-parachain-url",
+			"ws://localhost:9944",
```

Comment:
I assume we actually have to change these arguments because some of them are not supported by our vault client.